smurfs/src/actions/actions.js

```diff
@@ -0,0 +1,50 @@
+import axios from 'axios';
+//action types
+export const FETCH_SMURFS_START = 'FETCH_SMURFS_START';
+export const FETCH_SMURFS_SUCCESS = 'FETCH_SMURFS_SUCCESS';
+export const FETCH_SMURFS_FAILURE = 'FETCH_SMURFS_FAILURE';
+export const ADD_SMURFS = 'ADD_SMURFS';
+export const REMOVE_SMURFS = 'REMOVE_SMURFS';
+
+//action creator
+
+export const getSmurf = () => dispatch => {
+    dispatch({ type: FETCH_SMURFS_START });
+    axios.get('http://localhost:3333/smurfs')
+        .then(res => {
+            dispatch({ type: FETCH_SMURFS_SUCCESS, payload: res.data });
+        })
+        .catch(err => {
+            dispatch({ type: FETCH_SMURFS_FAILURE, payload: err.response })
+        })
+};
+
+export const addSmurf = smurf => dispatch => {
+    dispatch({ type: ADD_SMURFS });
+    axios.post('http://localhost:3333/smurfs', smurf)
+        .then(res => {
+            console.log(res.data)
+            dispatch({ type: ADD_SMURFS, payload: res.data })
+            dispatch({ type: FETCH_SMURFS_SUCCESS, payload: res.data })
```

Comment:
this would probably work better if you only used the FETCH_SMURFS_SUCCESS and got rid of the ADD_SMURFS dispatch. Right now ADD_SMURFS in your reducer is pretty much doing the same thing as your FETCH_SMURFS_SUCCESS so why write more code than you have to?